Engine/World.cs

```diff
@@ -96,6 +143,19 @@ public void RemoveEntity(Entity entity)
             _entitiesRemovedBuffer.Add(entity);
         }
 
+        /// <summary>
+        /// Updates the region of an entity.
+        /// </summary>
+        /// <param name="entity">Entity to be updated</param>
+        public void UpdateEntityRegion(Entity entity)
+        {
+            if (Entities.Contains(entity))
+            {
+                RemoveEntity(entity);
+            }
+            AddEntity(entity);
```

Comment:
This will trigger EntityAdded and EntityRemoved all the time since these methods should only be used for the creation and destruction of the entity.

A more optimized and correct way could be to store in each entity its current region's coordinates to directly know from what region it should be removed and directly add it in the new correct region.
What we have to do with this technique is doing the same thing I did with the add/removal of entities actually and make a "region add/remove buffer" to keep track of which entities needs to be added/removed at the end of the current tick.

Reply:
Done (I actually didn't see the second comment before asking you on discord, whoops!)